cloudformation/cid-crdc-resources.yaml

```diff
@@ -0,0 +1,709 @@
+              execute_query('ALTER TABLE {table_name} DROP IF EXISTS PARTITION {partition}'.format(
+                  table_name=TABLE_NAME,
+                  partition=build_partition_string(accountid_partition_value, region_partition_value, dt_partition_value)))
+                  
+          def build_partition_string(accountid_partition_value, region_partition_value, dt_partition_value):
+              return "(accountid='{accountid_partition_value}', dt='{dt_partition_value}', region='{region_partition_value}')".format(
+                accountid_partition_value=accountid_partition_value,
+                  dt_partition_value=dt_partition_value,
+                  region_partition_value=region_partition_value)
+
+          def execute_query(query):
+              print('Executing query:', query)
+              query_output_location = 's3://{athena_query_results_bucket}'.format(
+                  athena_query_results_bucket=ATHENA_QUERY_RESULTS_BUCKET_NAME,
+                  account_id=ACCOUNT_ID,
+                  region=ATHENA_REGION)
+              start_query_response = athena.start_query_execution(
+                  QueryString=query,
+                  QueryExecutionContext={
+                      'Database': ATHENA_DATABASE_NAME
+                  },
+                  ResultConfiguration={
+                      'OutputLocation': query_output_location,
+                  },
+                  WorkGroup=ATHENA_WORKGROUP
+              )
+              print('Query started')
+              
+              is_query_running = True
+              while is_query_running:
+                  get_query_execution_response = athena.get_query_execution(
+                      QueryExecutionId=start_query_response['QueryExecutionId']
+                  )
+                  query_state = get_query_execution_response['QueryExecution']['Status']['State']
+                  is_query_running = query_state in ('RUNNING', 'QUEUED')
+                  
+                  if not is_query_running and query_state != 'SUCCEEDED':
+                      raise AthenaException('Query failed')
+              print('Query completed')
+
+  IAMRoleLambdaPartitionerConfig:
+    Type: AWS::IAM::Role
+    DeletionPolicy: "Delete"
+    UpdateReplacePolicy: "Delete"
+    Description: "CID-CRCD Dashboard - Allows to add partitions to Athena and Glue, send logs to Cloudwatch, access Athena query results S3 bucket, receive objects from Config bucket. Each defined in separate policies"
+    Properties:
+      Path: "/"
+      ManagedPolicyArns:
+      - Ref: "IAMManagedPolicyLambdaAthena"
+      - Ref: "IAMManagedPolicyLambdaGlue"
+      - Ref: "IAMManagedPolicyLambdaS3AthenaQueryResults"
+      - Ref: "IAMManagedPolicyLambdaCloudWatchLogs"
+      - Ref: "IAMManagedPolicyLambdaS3ConfigSnapshotObject"
+      MaxSessionDuration: 3600
+      RoleName: !Sub "cid-crcd-lambda-partitioner-role-${AWS::Region}-${AWS::AccountId}"
```

Comment:
Role Name typically do not need regions nor account id suffix

Reply:
done

Reply:
resolving as agreed with Iakov